services/ai_client.py

```
@@ -5,37 +5,127 @@
+    def _validate_token_limits(self, system: str, prompt: str, max_tokens: int) -> None:
+        """Validate that the request doesn't exceed token limits."""
+        if not self.validate_tokens:
+            return
+            
+        input_tokens = self._count_tokens(system + prompt)
+        total_tokens = input_tokens + max_tokens
+        
+        if input_tokens > self.max_input_tokens:
+            raise TokenLimitExceededError(
+                f"Input tokens ({input_tokens:,}) exceed model limit ({self.max_input_tokens:,})"
+            )
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)